cmake/faceliftMacros.cmake

```diff
@@ -388,6 +390,30 @@ macro(_facelift_parse_target_arguments additionalOptions additionalOneValueArgs
         list(APPEND HEADERS ${GLOB_FILES})
     endforeach()
 
+    set(HEADERS_NO_MOC ${ARGUMENT_HEADERS_NO_MOC})
+
+    # files for installation CANNOT be in both sections: HEADERS and HEADERS_NO_MOC
+    foreach(HEADER ${HEADERS})
+        string(FIND ";${HEADERS_NO_MOC}" ";${HEADER}" IS_FOUND )
+        if(NOT ${IS_FOUND} LESS 0)
+            message("HEADERS_NO_MOC is '${HEADERS_NO_MOC}'")
+            message("HEADERS is '${HEADERS}'")
+            message("HEADER is '${HEADER}'")
+            message(FATAL_ERROR "Header '${HEADER}' is present in both HEADERS and HEADERS_NO_MOC sections at position '${IS_FOUND}'")
+        endif()
+    endforeach()
+
+    UNSET(HEADERS_NO_MOC_GLOB)
```

Comment:
HEADERS_GLOB_NO_MOC vs HEADERS_NO_MOC_GLOB? It is confusing. Please improve naming.

Reply:
Agree!
lets HEADERS_GLOB_NO_MOC -> HEADERS_NO_MOC_GLOB etc
be for consequence:
ORDINARY | GLOB | GLOB RECURSE
-- | -- | --
HEADERS | HEADERS_GLOB | HEADERS_GLOB_RECURSE
SOURCES | SOURCES_GLOB | SOURCES_GLOB_RECURSE
HEADERS_NO_MOC | HEADERS_NO_MOC_GLOB | HEADERS_NO_MOC_GLOB_RECURSE

Facelift without extra moc processing | Current Github version
-- | --
1:27 | 1:48
1:03 | 1:05
1:20 | 1:47
1:30 | 1:08
1:10 | 1:56
0:49 | 1:07
1:03 | 1:23
1:17 | 1:33
0:52 | 1:01